packages/serverpod_serialization/lib/src/auth_encoding.dart

```diff
@@ -0,0 +1,62 @@
+/// Serverpod's default authentication key encoding and decoding.
+library serverpod_serialization.auth_encoding;
+
+import 'dart:convert';
+
+/// The name of the default Serverpod scheme for HTTP "authorization" headers.
+/// Note, the scheme name is case-insensitive and should be compared in a case-insensitive manner.
+const serverpodDefaultAuthSchemeName = 'Basic';
+
+/// Regexp for a non-empty string adhering to the RFC 4648 base64 encoding alphabet.
+const _base64RegExpStr = r'[a-zA-Z0-9+/]*=*';
+
+/// The regular expression for the format of the authentication scheme.
+const _authSchemeRegExpStr = r'[a-zA-Z0-9_-]+';
+
+/// The regular expression for the format of the value in an HTTP header.
+/// This also applies to the authentication data payload / key.
+/// It consists of:
+/// - \t (tab character)
+/// - \x20-\x7E (printable ASCII characters, including space)
+/// - \x80-\xFF (extended ASCII characters)
+const _httpHeaderValueRegExpStr = r'[\t\x20-\x7E\x80-\xFF]*';
+
+/// The regular expression for the format of the HTTP "authorization" header value,
+/// including starting with an authentication scheme name.
+final _authValueRegExp =
+    RegExp('^$_authSchemeRegExpStr $_httpHeaderValueRegExpStr\$');
+
+/// Returns true if the provided value is a valid HTTP "authorization" header value
+/// (which includes starting with an authentication scheme name).
+bool isValidAuthHeaderValue(String value) => _authValueRegExp.hasMatch(value);
+
+/// The regular expression for the format of Serverpod-wrapped auth keys.
+final _wrappedAuthValueRegExp =
+    RegExp('^$serverpodDefaultAuthSchemeName $_base64RegExpStr\$');
+
+/// Returns true if the provided value is a Serverpod-wrapped auth key.
+bool isWrappedAuthValue(String value) =>
+    _wrappedAuthValueRegExp.hasMatch(value);
+
+/// Returns a value that is compliant with the HTTP auth header format
+/// by encoding and wrapping the provided auth key as a Basic auth value.
+String wrapAuthValue(String key) {
```

Comment:
Question: Since this is a header compliant format, I'm wondering if it would make sense to name this something very explicit like `createHttpAuthHeader`. Right now it's a bit more opaque, but perhaps there's some benefit to that?

The same reasoning applies to `isWrappedAuthValue`, which is (as far as I can tell?) equivalent to `isHttpAuthHeader` (or maybe `isDefaultHttpAuthHeader` specifically)?

Reply:
This code is designed to also support the case where users implement their own authorization approach, potentially using some other HTTP auth scheme, e.g. Bearer. It is only in the Basic case we actually perform a modification (wrapping).

Reply:
Ah, I see! But I'm still not quite following why we have to be opaque/generic with the naming and not state that we're working with HTTP headers? Maybe I'm missing the bigger picture here 😊

Reply:
Ok I've made it a bit more explicit with wrapAuthHeaderValue!